bundles/org.openhab.binding.roborock/src/main/java/org/openhab/binding/roborock/internal/RoborockVacuumHandler.java

```diff
@@ -0,0 +1,1015 @@
+                    if (home != null) {
+                        rrHomeId = Integer.toString(home.data.rrHomeId);
+                    }
+                }
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                        "@text/offline.conf-error.no-token");
+            }
+        }
+
+        scheduleNextPoll(-1);
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private synchronized void teardown(boolean scheduleReconnection) {
+        pollTask.cancel();
+        initTask.cancel();
+
+        if (scheduleReconnection) {
+            initTask.schedule(30);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        teardown(false);
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("Bridge status changed to {}", bridgeStatusInfo);
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            initTask.submit();
+        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+            teardown(false);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    private void updateDevice(Devices devices[]) {
+        for (int i = 0; i < devices.length; i++) {
+            if (config.duid.equals(devices[i].duid)) {
+                if (localKey.isEmpty()) {
+                    localKey = devices[i].localKey;
+                }
+                updateState(CHANNEL_ERROR_ID, new DecimalType(devices[i].deviceStatus.errorCode));
+                updateState(CHANNEL_STATE_ID, new DecimalType(devices[i].deviceStatus.vacuumState));
+                updateState(CHANNEL_BATTERY, new DecimalType(devices[i].deviceStatus.battery));
+                updateState(CHANNEL_FAN_POWER, new DecimalType(devices[i].deviceStatus.fanPower));
+                updateState(CHANNEL_MOP_DRYING, new DecimalType(devices[i].deviceStatus.dryingStatus));
+                updateState(CHANNEL_CONSUMABLE_MAIN_PERC, new DecimalType(devices[i].deviceStatus.mainBrushWorkTime));
+                updateState(CHANNEL_CONSUMABLE_SIDE_PERC, new DecimalType(devices[i].deviceStatus.sideBrushWorkTime));
+                updateState(CHANNEL_CONSUMABLE_FILTER_PERC, new DecimalType(devices[i].deviceStatus.filterWorkTime));
+
+                if (devices[i].online) {
+                    sendAllMqttCommands();
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "@text/offline.comm-error.vac-offline");
+                }
+            }
+        }
+    }
+
+    private void pollData() {
+        logger.debug("Running pollData for: {}", config.duid);
+        HomeData homeData = bridgeHandler.getHomeData();
+        if ((homeData != null && (homeData.result != null))) {
```

Comment:
Simplify. Java checks per condition and falls thru early
```suggestion
        if (homeData != null && homeData.result != null) {
```

Reply:
Done.